### PR TITLE
[CPU] BitwiseLeftShift & BitwiseRightShift CPU implementation

### PR DESCRIPTION
**Pull request by mitruska**
### Details:
 - BitwiseLeftShift & BitwiseRightShift CPU base implementation 
 - [Update] Added dedicated CPU tests class for BitwiseShift ops to specify data
 -----------------------------
 - Related PR:
  https://github.com/openvinotoolkit/openvino/pull/25857

### Tickets:
 - 148536

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@nshchego could you please take a look from CPU side

File: src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/instances/x64/eltwise.cpp [hunk 1]
```diff
@@ -750,6 +750,51 @@ const auto params_4D_bitwise_NOT_i32 = ::testing::Combine(
 
 INSTANTIATE_TEST_SUITE_P(smoke_CompareWithRefs_4D_Bitwise_NOT_i32, EltwiseLayerCPUTest, params_4D_bitwise_NOT_i32, EltwiseLayerCPUTest::getTestCaseName);
 
+const auto params_4D_bitwise_shift = ::testing::Combine(
```
Comment (nshchego) [inline src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/instances/x64/eltwise.cpp:753]: Please move these test cases to src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/instances/common/eltwise.cpp, due to the current implementation should work on all platforms.

File: src/plugins/intel_cpu/tests/functional/shared_tests_instances/skip_tests_config.cpp [hunk 2]
```diff
@@ -308,6 +308,8 @@ std::vector<std::string> disabledTestPatterns() {
         R"(.*smoke_TestsROIAlign.*)",
         // Issue: 148527
         R"(.*Snippets.*MatMulTransposeB.*i8.*i8.*)",
+        // Issue: 136881
+        R"(.*smoke_CompareWithRefs_4D_Bitwise.*/EltwiseLayerCPUTest.*_eltwise_op_type=BitwiseLeft.*_model_type=.*(i16|u16|u32).*)",
```
Comment (nshchego) [inline src/plugins/intel_cpu/tests/functional/shared_tests_instances/skip_tests_config.cpp:312]: So, does the Right shift works for these ETs? It's confusing. The logical right shift should not work. You may need to change the generated data to cover such cases.

**Review by rkazants — APPROVED** [review]
LGTM from TF FE point

File: src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/instances/x64/eltwise.cpp (hunk 1 above)
Comment (mitruska) [reply src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/instances/x64/eltwise.cpp]: Moved to the common (src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/instances/common/bitwise_shift.cpp)

File: src/plugins/intel_cpu/tests/functional/shared_tests_instances/skip_tests_config.cpp (hunk 2 above)
Comment (mitruska) [reply src/plugins/intel_cpu/tests/functional/shared_tests_instances/skip_tests_config.cpp]: I've added class to test the Bitwise Shift ops with custom data and more cases.
In general the RightShift doesn't fail for the same edge cases when the sign bit or max value conversion is not involved.

File: src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/bitwise_shift.cpp
Comment (mitruska) [inline src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/bitwise_shift.cpp:1]: [Update] Added dedicated CPU BitwiseShift test class to provide custom data, and generate more specific cases.
It's based on the tests for eltwise ops test code and params (src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/eltwise.cpp)

Further improvements to the tests can be provided as a follow up if needed.

**Review by mmikolajcz — APPROVED** [review]

File: src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/bitwise_shift.cpp
Comment (mitruska) [reply src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/bitwise_shift.cpp:1]: There is suitable shared_tests class "eltwise" that can be reused, so I've instantiated them as well (it has been already used for GPU Bitwise ops): https://github.com/openvinotoolkit/openvino/blob/ce92c13f05f0a9e40fc094ae04f20ceafbd8df19/src/plugins/intel_gpu/tests/functional/shared_tests_instances/single_layer_tests/eltwise.cpp#L69-L70

For the custom CPU tests I used BitwiseAnd/Or as an example (PR https://github.com/openvinotoolkit/openvino/pull/20474), the "ref" type from CPUSpecificParams is checked, and in fact they can be reused to test JIT impl when implemented. 
We can discuss offline this test class location and improvements, so I can apply them without blocking the Bitshift ops enablement.

File: src/plugins/intel_cpu/src/nodes/eltwise.cpp [hunk 3]
```diff
@@ -3108,7 +3131,9 @@ bool Eltwise::canFuse(const NodePtr& node) const {
                Algorithm::EltwiseBitwiseAnd,
                Algorithm::EltwiseBitwiseNot,
                Algorithm::EltwiseBitwiseOr,
-               Algorithm::EltwiseBitwiseXor)) {
+               Algorithm::EltwiseBitwiseXor,
+               Algorithm::EltwiseBitwiseLeftShift,
+               Algorithm::EltwiseBitwiseRightShift)) {
```
[the remark this answers is not in the document]
Comment (mitruska) [reply src/plugins/intel_cpu/src/nodes/eltwise.cpp:3138]: Ok, applied. Thanks!

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**